consensus/src/quorum_store/proof_coordinator.rs

```diff
@@ -116,7 +123,9 @@ impl ProofCoordinator {
         validator_verifier: &ValidatorVerifier,
     ) -> Result<Option<ProofOfStore>, SignedDigestError> {
         if !self.digest_to_proof.contains_key(&signed_digest.digest()) {
-            if signed_digest.info().batch_author == self.peer_id {
+            if signed_digest.info().batch_author == self.peer_id
```

Comment:
I think we missed a possible attack. I can make you store incremental signatures for all digests in the system because I can lie about the batch_author in the signed digest. I think the fix is to check signed_digest.info().batch_author against what we store. Probably require an API change to batch_reader.exists.

Reply:
but in order to check we should be storing it, no? so I guess that's why the original algorithm initialized the proofs from self (assuming self is and should be faster) and then appended matching digest/signatures - and I guess that also explains the check of the own signature, but in this case it should have been an assert.

I think that's a bit nicer than trying to check the author - as long as I'm not mistaken and checking the author almost implies that checking own signature would also be okay.

Reply:
@igor-aptos